payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncher.kt

```diff
@@ -104,28 +97,50 @@ class GooglePayLauncher internal constructor(
         readyCallback: ReadyCallback,
         resultCallback: ResultCallback
     ) : this(
+        applicationContext = fragment.requireContext().applicationContext,
         lifecycleScope = fragment.lifecycleScope,
         config = config,
         readyCallback = readyCallback,
-        activityResultLauncher = fragment.registerForActivityResult(
+        launcher = fragment.registerForActivityResult(
             GooglePayLauncherContract(),
             resultCallback::onResult,
         ),
+        analyticsRequestFactory = PaymentAnalyticsRequestFactory(
+            fragment.requireContext(),
+            PaymentConfiguration.getInstance(fragment.requireContext()).publishableKey,
+            setOf(PRODUCT_USAGE)
+        ),
+        analyticsRequestExecutor = DefaultAnalyticsRequestExecutor()
+    )
+
+    internal constructor(
+        applicationContext: Context,
+        lifecycleScope: CoroutineScope,
+        launcher: ActivityResultLauncher<GooglePayLauncherContract.Args>,
+        analyticsRequestFactory: PaymentAnalyticsRequestFactory,
+        analyticsRequestExecutor: AnalyticsRequestExecutor,
+        config: Config,
+        readyCallback: ReadyCallback,
+    ) : this(
+        lifecycleScope = lifecycleScope,
+        config = config,
+        readyCallback = readyCallback,
+        activityResultLauncher = launcher,
         googlePayRepositoryFactory = {
             DefaultGooglePayRepository(
-                context = fragment.requireActivity().application,
+                context = applicationContext,
                 environment = config.environment,
                 billingAddressParameters = config.billingAddressConfig.convert(),
                 existingPaymentMethodRequired = config.existingPaymentMethodRequired,
-                allowCreditCards = config.allowCreditCards
+                allowCreditCards = config.allowCreditCards,
+                errorReporter = RealErrorReporter(
+                    analyticsRequestExecutor = analyticsRequestExecutor,
+                    analyticsRequestFactory = analyticsRequestFactory,
+                )
             )
         },
-        paymentAnalyticsRequestFactory = PaymentAnalyticsRequestFactory(
-            context = fragment.requireContext(),
-            publishableKey = PaymentConfiguration.getInstance(fragment.requireContext()).publishableKey,
-            defaultProductUsageTokens = setOf(PRODUCT_USAGE),
-        ),
-        analyticsRequestExecutor = DefaultAnalyticsRequestExecutor(),
+        paymentAnalyticsRequestFactory = analyticsRequestFactory,
+        analyticsRequestExecutor = analyticsRequestExecutor,
```

Comment:
Created this constructor in order to avoid recreating `PaymentAnalyticsRequestFactory` and `AnalyticsRequestExecutor`. It also ended up moving all the logic for building `DefaultGooglePayRepository` into a single location.